### PR TITLE
Student course details pages

### PR DESCRIPTION
**Pull request by SonaliBedge**
This PR is built on top of all the other PRs—PR #36, PR #32, PR #29, and PR #25. I have merged all the branches—QuizPage, LessonPage, SignUpAndLoginPage, and LearnsLessonsPage—into the StudentCourseDetailsPages branch. All the latest changes are now available in the StudentCourseDetailsPages branch.

### CODE REVIEW

File: server/models/CourseModel.mjs
Comment (OGoncharenko) [inline server/models/CourseModel.mjs:1]: Instead of commenting some of the lines I can suggest to change them this way:
`    // Validate types
    if (typeof courseId !== 'string') throw new Error('Invalid courseId');
    if (typeof courseName !== 'string') throw new Error('Invalid courseName');
    if (typeof description !== 'string') throw new Error('Invalid description');
    if (typeof imageUrl !== 'string') throw new Error('Invalid imageUrl');
    if (typeof createdBy !== 'string') throw new Error('Invalid createdBy');
    if (!(createdAt instanceof Date)) throw new Error('Invalid createdAt');

    // Optional fields
    if (courseType && typeof courseType !== 'string') throw new Error('Invalid courseType');
    if (logoUrl && typeof logoUrl !== 'string') throw new Error('Invalid logoUrl');
    if (duration && typeof duration !== 'string') throw new Error('Invalid duration');
    if (rating && typeof rating !== 'number') throw new Error('Invalid rating');
    if (otherInfo && typeof otherInfo !== 'string') throw new Error('Invalid otherInfo');`

This changes to resolve all errors.
I made this changes in my PR #34. You can add this simple changes and push here if you want

File: src/components/SideBar/SideBar.jsx
Comment (OGoncharenko) [inline src/components/SideBar/SideBar.jsx:1]: I believe these changes in the main already

File: src/components/SideBar/SideBar.module.css
Comment (OGoncharenko) [inline src/components/SideBar/SideBar.module.css:1]: And these changes are in the main

File: src/components/SideBarItems/SideBarItemStyles.module.css
Comment (OGoncharenko) [inline src/components/SideBarItems/SideBarItemStyles.module.css:1]: Same here, these changes are in the main

File: src/components/SideBarItems/SideBarItems.jsx
Comment (OGoncharenko) [inline src/components/SideBarItems/SideBarItems.jsx:1]: and here

**Comment by OGoncharenko** [comment]
I would advice you to rebase main into your branch to resolve all merge conflicts. I see a lot of my code changes that are already in main branch.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (OGoncharenko) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:1]: TeacherDashboard is not related to a Student course details pages and could not be included in this PR.

File: src/pages/TeacherDashboard/TeacherDashboard.module.css
Comment (OGoncharenko) [inline src/pages/TeacherDashboard/TeacherDashboard.module.css:1]: The same here, changes are already in main branch

**Review by OGoncharenko — CHANGES_REQUESTED** [review]

File: server/models/CourseModel.mjs
Comment (SonaliBedge) [reply server/models/CourseModel.mjs:1]: Got it!

File: src/components/SideBar/SideBar.jsx
Comment (SonaliBedge) [reply src/components/SideBar/SideBar.jsx:1]: before doing any change I got all changes from main only. I don't know why its showing. I didn't teach SideBar.jsx file code.

File: src/components/SideBar/SideBar.module.css
Comment (SonaliBedge) [reply src/components/SideBar/SideBar.module.css:1]: same here, before doing any change I got all changes from main only. I don't know why its showing. I didn't teach SideBar.module.jsx file code.

File: src/components/SideBarItems/SideBarItemStyles.module.css
Comment (SonaliBedge) [reply src/components/SideBarItems/SideBarItemStyles.module.css:1]: same here, before doing any change I got all changes from main only. I don't know why its showing. I didn't teach SideBarItemStyles.module.css file code.

File: src/components/SideBarItems/SideBarItems.jsx
Comment (SonaliBedge) [reply src/components/SideBarItems/SideBarItems.jsx:1]: same here, before doing any change I got all changes from main only. I don't know why its showing. I didn't teach SideBarItems.jsx file code.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (SonaliBedge) [reply src/pages/TeacherDashboard/TeacherDashboard.jsx:1]: I didn't include it, while doing rebase its got added. I am not good in rebasing.

File: src/pages/TeacherDashboard/TeacherDashboard.module.css
Comment (SonaliBedge) [reply src/pages/TeacherDashboard/TeacherDashboard.module.css:1]: same here too

Comment (EdithHarrison) [reply src/pages/TeacherDashboard/TeacherDashboard.module.css:1]: Since this branch was made on top of other branches and some of these branches are merged in the main now, you will have a lot of conflict. I suggest you do git rebase origin/main to fix this conflicts.

Comment (SonaliBedge) [reply src/pages/TeacherDashboard/TeacherDashboard.module.css:1]: I resolved all conflicts.
